core/src/test/java/org/testcontainers/DockerRegistryContainer.java

```diff
@@ -0,0 +1,68 @@
+package org.testcontainers;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.FrameConsumerResultCallback;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.output.WaitingConsumer;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DockerRegistryContainer extends GenericContainer<DockerRegistryContainer> {
+
+    @Getter
+    String endpoint;
+
+    public DockerRegistryContainer() {
+        super(TestImages.DOCKER_REGISTRY_IMAGE);
+    }
+
+    public DockerRegistryContainer(@NonNull Future<String> image) {
+        super(image);
+    }
+
+    @Override
+    protected void configure() {
+        super.configure();
+        withEnv("REGISTRY_HTTP_ADDR", "127.0.0.1:0");
+        withCreateContainerCmdModifier(cmd -> {
+            cmd.getHostConfig().withNetworkMode("host");
+        });
+    }
+
+    @Override
+    @SneakyThrows
+    protected void containerIsStarting(InspectContainerResponse containerInfo) {
+        AtomicInteger port = new AtomicInteger(-1);
+        try (FrameConsumerResultCallback resultCallback = new FrameConsumerResultCallback()) {
+            WaitingConsumer waitingConsumer = new WaitingConsumer();
+            resultCallback.addConsumer(OutputFrame.OutputType.STDERR, waitingConsumer);
+
+            dockerClient.logContainerCmd(containerInfo.getId())
+                .withStdErr(true)
+                .withFollowStream(true)
+                .exec(resultCallback);
+
+            Pattern pattern = Pattern.compile(".*listening on .*:(\\d+).*", Pattern.DOTALL | Pattern.CASE_INSENSITIVE | Pattern.MULTILINE);
```

Comment:
Pattern creation could move out of the `try`, or?

Reply:
this code was moved from another class. So, unless there is a problem with it being here, I would rather not change anything, as the focus is on something else.

FTR I think it makes absolutely no sense to move it, as long as we are not compiling it many times (eg in a loop or callback)

Reply:
You are right, I looked wrongly and thought it was in a retry block.